CRM/Admin/Page/Extensions.php

```diff
@@ -157,6 +157,10 @@ public function formatLocalExtensionRows() {
         continue;
       }
 
+      if (isset($obj->ui_visibility) && !$obj->ui_visibility === FALSE) {
+        continue;
+      }
+
```

Comment:
1. May I suggest #16551 and something like `in_array('hidden', $obj->tags)`?

2. I like the idea of adding more metadata and using that to influence the extension-management. For example: which extensions are activated automatically during installation? Which can be disabled or uninstalled? 

3. The phrase "ui visibility" feels odd to me. Does that convey something more than `visible` or `hidden` would convey?

I'd sort of like to see a list of extension-management-related tags that we can put into dev-docs. Spitballing:

* `autoinstall`: The extension is activated during Civi installation)
* `hidden`: The extension should omitted from the admininistrative GUI (but still be visible to APIs)
* `mandatory`: The extension cannot be disabled. (This may also imply that it's activated during Civi installation.)
* `suggested`: The extension should be presented as an option in the install UI (alongside the various components)

<img width="1166" alt="Screen Shot 2020-02-14 at 6 03 12 PM" src="https://user-images.githubusercontent.com/1336047/74579999-529ba280-4f54-11ea-9b01-beeb2a0f199e.png">

Reply:
Ohh I like the look of that install proposal!

I'm gonna  move to that  PR to comment

Reply:
Updated above^^